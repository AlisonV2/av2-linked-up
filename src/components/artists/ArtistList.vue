<template>
  <div class="row">
    <div
      class="col-8 col-md-6 col-lg-4 col-xl-3 artists-list"
      v-for="artist in artists"
      :key="artist.name"
    >
      <router-link :to="{ name: 'Profile', params: { id: artist.id } }">
        <ArtistItem :artist="artist" />
      </router-link>
    </div>
  </div>
</template>

<script>
import ArtistItem from '@/components/artists/ArtistItem';
import * as Sentry from '@sentry/vue';

/**
 * @exports ArtistList
 * @type {Component}
 * @vue-data {array} artists
 * @vue-event created - Dispatch store action
 */
export default {
  name: 'ArtistList',
  components: {
    ArtistItem,
  },
  data() {
    return {
      artists: [],
    };
  },
  /**
   * @description Dispatch getAllArtists action on created hook
   * Get results from store
   * @method created
   * @returns {array}
   * @async
   */
  async created() {
    await this.$store.dispatch('getAllArtists')
    .then(() => {
      this.artists = this.$store.getters.getAllArtists;
    })
    .catch((err) => Sentry.captureException(err));
  },
};
</script>
