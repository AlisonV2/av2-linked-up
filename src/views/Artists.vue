<template>
  <div class="container">
    <div class="row">
      <app-title>Find your artist</app-title>
    </div>
      <ArtistList />
  </div>
</template>

<script>
import ArtistList from '@/components/artists/ArtistList';

export default {
  name: 'Artists',
  components: {
    ArtistList,
  },
};
</script>
