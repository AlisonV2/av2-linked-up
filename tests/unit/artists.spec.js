import { shallowMount } from '@vue/test-utils';
import Artists from '@/views/Artists.vue';
import ArtistList from '@/components/artists/ArtistList.vue';
import AppTitle from '@/components/app/Title.vue';

const config = {
  global: {
    components: {
      ArtistList,
      'app-title': AppTitle,
    },
  },
};

/**
 * @module ArtistPageTest
 */
describe('Artists.vue', () => {
  it('Artists Snapshot', () => {
    const component = shallowMount(Artists, config);
    expect(component.element).toMatchSnapshot();
  });

  it('Check if ArtistList is rendered', () => {
    const component = shallowMount(Artists, config);
    const artists = component.findComponent(ArtistList);
    expect(artists.exists()).toBe(true);
  });
});
