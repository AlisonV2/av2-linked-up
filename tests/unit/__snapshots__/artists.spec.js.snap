// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artists.vue Artists Snapshot 1`] = `
<div
  class="container"
>
  <div
    class="row"
  >
    <app-title-stub
      mode="title-img"
    />
  </div>
  <artist-list-stub />
</div>
`;
